libs/utils/utils.js

```diff
@@ -655,17 +666,47 @@ async function loadFooter() {
   await loadBlock(footer);
 }
 
+function decorateSection(section, idx) {
+  let links = decorateLinks(section);
+  decorateDefaults(section);
+  const blocks = section.querySelectorAll(':scope > div[class]:not(.content)');
+
+  const { doNotInline } = getConfig();
+  const blockLinks = [...blocks].reduce((blkLinks, block) => {
+    const blockName = block.classList[0];
+    links.forEach((link) => {
+      if (!block.contains(link)) return;
```

Comment:
nit: 
I'd recommend filtering first so you don't need that `if` in the beginning -> links.filter((link) => 
`links.filter((link) => block.contains(link)).forEach((link) => { ...`

just seems a little nicer